Finalize and publish release in CI + various automation updates

mokagio opened the pull request:
- Use `RELEASE_VERSION` at the pipeline YAML level instead of reading `BUILDKITE_RELEASE_VERSION` in the script.
- Add pipelines for the release finalization and release publishing steps + `publish_release` lane
- Update release-toolkit and adopt `buildkite_pipeline_upload`
- Adopt DRY `create_backmerge_prs!` from Simplenote Android
- Isolate logic to close GitHub milestone in retry-able way so we can reuse it in other projects

mokagio committed:
Use latest release-toolkit

mokagio committed:
Refine `checkout-release-branch.sh` to use `$1` only

We don't want to use the `BUILDKITE_` prefix workaround.

mokagio committed:
DRY "Configure Git for..." Buildkite header

mokagio committed:
Adopt DRY create_backmerge_prs! from Simplenote Android

mokagio committed:
Tweak hotfix usage message

mokagio committed:
Use dash over dots for simpler text manipulation

mokagio committed:
Remove call to create new milestone – Rv2 does that

mokagio committed:
Extract logic to close milestone

mokagio committed:
Add lane to publish release on GitHub

mokagio committed:
Add pipeline for release finalization

mokagio committed:
Add pipeline to publish release

mokagio committed:
Add pipeline to upload metadata on ASC

mokagio committed:
Set release-toolkit version as 12.2 explicitly to use new feature

This way, we'll always be sure the gems resolved to a version that
includes this new feature we want to use: `buildkite_pipeline_upload`

mokagio committed:
Add `UI.success` to method to trigger Buildkite build

mokagio committed:
Update trigger build logic to upload pipeline when in CI

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR has the <code>Releases...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR has the <code>Releases</code> label: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Update CocoaPods and AppCenter to work with Xcode 16.1

The previous AppCenter version failed the build with a "rsync.samba"
error due to it having script sandboxing enabled.

Noticed that:

- The only breaking change from version 4 to 5 is the change in minimum
  deployment target. See
  https://github.com/microsoft/appcenter-sdk-apple/blob/8b039c9de609bd58ea7526c60954194cd1b3a94e/CHANGELOG.md#version-500
- The App Store build does not import AppCenter
- AppCenter will be decommissioned in March 2025

mokagio commented:
The UI tests failed. But that cannot be due to the AppCenter version changes (and of course the release automation) because the build that runs the UI tests does not import AppCenter.
